mlflow/pyfunc/__init__.py

```diff
@@ -127,13 +137,29 @@ def load_pyfunc(path, run_id=None):
         raise Exception("Format '{format}' not found not in {path}.".format(format=FLAVOR_NAME,
                                                                             path=conf_path))
     conf = model.flavors[FLAVOR_NAME]
+    model_py_version = conf.get(PY_VERSION)
+    if not suppress_warnings:
+        _warn_potentially_incompatible_py_version_if_necessary(model_py_version=model_py_version)
     if CODE in conf and conf[CODE]:
         code_path = os.path.join(path, conf[CODE])
         sys.path = [code_path] + _get_code_dirs(code_path) + sys.path
     data_path = os.path.join(path, conf[DATA]) if (DATA in conf) else path
     return importlib.import_module(conf[MAIN]).load_pyfunc(data_path)
 
 
+def _warn_potentially_incompatible_py_version_if_necessary(model_py_version):
+    if model_py_version is None:
+        eprint("The specified model does not have a specified Python version. It may be"
+              " incompatible with the version of Python that is currently running:"
+              " Python {version}".format(
+                   version=PYTHON_VERSION))
+    elif get_major_minor_py_version(model_py_version) != get_major_minor_py_version(PYTHON_VERSION):
+        eprint("The version of Python that the model was saved in, Python {model_version}, differs"
+              " from the version of Python that is currently running, Python {system_version},"
```

Comment:
lint will complain about this and the next line being one space off; as well as the eprint change above.